_api-reference/ingest-apis/ip2geo.md

```diff
@@ -0,0 +1,247 @@
+* GeoLite2 Country: https://geoip.maps.opensearch.org/v1/geolite2-country/manifest.json
+* GeoLite2 ASN: https://geoip.maps.opensearch.org/v1/geolite2-asn/manifest.json
+
+If OpenSearch cannot update a data source from the endpoints in 30 days, OpenSearch does not add GeoIP data to the documents, instead it adds `"error":"ip2geo_data_expired"`.
+
+The following table lists the IP2Geo data source options.
+
+| Name | Required | Default | Description |
+|------|----------|---------|-------------|
+| endpoint | no | https://geoip.maps.opensearch.org/v1/geolite2-city/manifest.json | The endpoint for downloading the GeoIP data. |
+| update_interval_in_days | no | 3 | The frequency in days for updating the GeoIP data; minimum value is 1. |
+
+The following code example shows how to create an IP2Geo data source.
+
+#### Example: PUT request
+
+```json
+PUT /_plugins/geospatial/ip2geo/datasource/my-datasource
+{
+    "endpoint" : "https://geoip.maps.opensearch.org/v1/geolite2-city/manifest.json",
+    "update_interval_in_days" : 3
+}
+```
+
+The following code example shows the reponse to the preceding request. A true response means the request was successful and the server was able to process the request. A false reponse means check the request to make sure it is valid, check the URL to make sure it is correct, or try again.
+
+#### Example: Successful response
+
+```json
+{
+    "acknowledged":true
+}
+```
+
+## Sending a GET request
+
+To get information about one or more IP2Geo data sources, send a GET request.  
+
+#### Example: GET request
+
+```json
+GET /_plugins/geospatial/ip2geo/datasource/my-datasource
+```
+
+#### Example: Response
+
+```json
+{
+  "datasources": [
+    {
+      "name": "my-datasource",
+      "state": "AVAILABLE",
+      "endpoint": "https://geoip.maps.opensearch.org/v1/geolite2-city/manifest.json",
+      "update_interval_in_days": 3,
+      "next_update_at_in_epoch_millis": 1685125612373,
+      "database": {
+        "provider": "maxmind",
+        "sha256_hash": "0SmTZgtTRjWa5lXR+XFCqrZcT495jL5XUcJlpMj0uEA=",
+        "updated_at_in_epoch_millis": 1684429230000,
+        "valid_for_in_days": 30,
+        "fields": [
+          "country_iso_code",
+          "country_name",
+          "continent_name",
+          "region_iso_code",
+          "region_name",
+          "city_name",
+          "time_zone",
+          "location"
+        ]
+      },
+      "update_stats": {
+        "last_succeeded_at_in_epoch_millis": 1684866730192,
+        "last_processing_time_in_millis": 317640,
+        "last_failed_at_in_epoch_millis": 1684866730492,
+        "last_skipped_at_in_epoch_millis": 1684866730292
+      }
+    }
+  ]
+}
+```
+
+## Updating an IP2Geo data source
+
+To update succesfully an IP2Geo data source, the GeoIP database from the latest-released database's endpoint must contain the same fields as those in the currently used database. Otherwise, the update fails. 
```

Comment:
```suggestion
To update an IP2Geo data source successfully, the GeoIP database from the new database's endpoint must contain all fields that the current database has. Otherwise, the update fails. 
```